src/dataframe.rs

```diff
@@ -539,3 +579,78 @@ fn print_dataframe(py: Python, df: DataFrame) -> PyResult<()> {
     print.call1((result,))?;
     Ok(())
 }
+
+fn project_schema(from_schema: Schema, to_schema: Schema) -> Result<Schema, ArrowError> {
+    let merged_schema = Schema::try_merge(vec![from_schema, to_schema.clone()])?;
+
+    let project_indices: Vec<usize> = to_schema
+        .fields
+        .iter()
+        .map(|field| field.name())
+        .filter_map(|field_name| merged_schema.index_of(field_name).ok())
+        .collect();
+
+    merged_schema.project(&project_indices)
+}
+
+fn record_batch_into_schema(
```

Comment:
I am surprised that `arrow-rs` nor `datafusion` have such a utility for converting a record-batch, but I did take a quick look around and didn't find anything.

Reply:
Well there is `cast`. Cast works on struct arrays, so you could make a simple wrapper around `cast` to work on `RecordBatch` by creating a struct array from the record batch. [This is what I do in pyo3-arrow](https://github.com/kylebarron/arro3/blob/c1cdb51527cb769694a4a400a8064a5e5a47290a/pyo3-arrow/src/ffi/to_python/utils.rs#L85-L87).

The main difference is that cast doesn't _also_ project. It's not clear to me whether the PyCapsule Interface intends to support projection or not. I don't think anyone has asked.

Reply:
Since the user isn't calling the pycapsule interface directly, it's also not clear how the user API would look to ask for a projection via pycapsules.